### DISCUSSION TITLE
Better error handling for when you don't wrap a t.field in quotes

### DISCUSSION THREAD

**Discussion by wakywayne** [hayes/pothos#640] [Ideas]
# I was trying to fix this code and kept getting this error:
## Here is my example: 

---

```
import { createServer } from '@graphql-yoga/node'
import SchemaBuilder from "@pothos/core"
import { CartItem, Cart, Money } from 'gql';

const builder = new SchemaBuilder({});


  //      Here is where we need better error handling, the only problem was that I didn't wrap *String* in quotes
        builder.objectType(Money, {
            name: "Money",
            description: "A money",
            fields: (t) => ({
                amount: t.exposeInt('amount', {}),
                formatted: t.field({
                    type: String,
                    resolve: (money) => new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD' }).format(money.amount),
                }),
            }),
        });


// This code works fine        

builder.objectType(Money, {
    name: "Money",
    description: "A money",
    fields: (t) => ({
        amount: t.exposeInt("amount", {}),
        formatted: t.field({
            type: "String",
            resolve: (money) =>
                new Intl.NumberFormat("en-US", {
                    style: "currency",
                    currency: "USD",
                }).format(money.amount),
        }),
    }),
});

```

### *I think the error message should also mention that "if you aren't using a ref function make sure your type is wrapped in quotes"*

**Comment by hayes** [comment]
This is an interesting case, the thing is you could technically use String as a valid ref for another type.  For most classes this error message would probably be more helpful.  We could potentially special case the "String" class and give a better error message, but it would be very specific to this exact typo

**Reply by wakywayne** [reply]
I just imagine in a large code base it's not that unlikely that somewhere quotes could be removed or neglected and if that is the case. You don't get a type error and the error message is useless. Could be a big headache for a relatively simple issue. I am a total noob though and don't understand any of the tradeoffs associated with solving this.
